File: n-queen/java/20221114/UserTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class UserTest {

    @Test
    void creation() {
        int n = 4;
        User user = new User(n);

        assertEquals(4, user.queens());
        assertEquals(new Board(n), user.board());
    }


    @Test
    void waysToPutQueens() {
        User user = new User(4);

        int ways =  user.findWaysToPutQueen();

        assertEquals(2, ways);
    }
}

File: 20230814-신고결과받기/src/UserTest.kt
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class UserTest {

    @Test
    fun `유저 당 한 번의 신고만 기록된다`() {
        val user = User("muzi")

        user.addReport("frodo")

        assertEquals(1, user.reportSize())

        user.addReport("frodo")

        assertEquals(1, user.reportSize())
    }
}
